### ISSUE TITLE
Error: Could not load UVM kernel module. Is nvidia-modprobe installed?

### ISSUE THREAD

**Issue by git3210** [NVIDIA/nvidia-docker#481]
journalctl -xe
Error: Could not load UVM kernel module. Is nvidia-modprobe installed?

modprobe nvidia_uvm
modprobe: FATAL: Module nvidia_uvm not found in directory /lib/modules/4.4.0-83-generic

how should i do?

**Comment by git3210** [comment]
/usr/bin/nvidia-docker-plugin -s $SOCK_DIR (code=exited, status=1/FAILURE)

**Comment by flx42** [comment]
Which distro are you using? And how did you install the NVIDIA drivers?

**Comment by git3210** [comment]
nvidia drivers install failed
error:
Unable to load the kernel module 'nvidia.ko'

i don't know how to deal this error

**Comment by illagrenan** [comment]
I had a similar issue on Ubuntu 16.04 LTS. I was able to install nvidia-docker prerequisites this way:

```bash
sudo add-apt-repository -y ppa:graphics-drivers/ppa
sudo apt update
sudo apt-get install -y nvidia-384 nvidia-modprobe

# Install Docker ...
# Install nvidia-docker ...
```

**Comment by flx42** [comment]
@illagrenan I don't recommend using this PPA, we package our drivers on our repository:
```
$ wget http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_9.0.176-1_amd64.deb
$ sudo dpkg -i cuda-repo-ubuntu1604_9.0.176-1_amd64.deb
```
This will add our repository:
```
$ cat /etc/apt/sources.list.d/cuda.list 
deb http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64 /
```

**Comment by dvenza** [comment]
I have the same error, but the module is loaded correctly:
```
$ lsmod | grep nvidia
nvidia_uvm            626688  0
nvidia_drm             53248  0
nvidia_modeset        790528  1 nvidia_drm
nvidia              12312576  11 nvidia_modeset,nvidia_uvm
drm_kms_helper        151552  1 nvidia_drm
drm                   360448  3 drm_kms_helper,nvidia_drm
$ sudo nvidia-docker run --rm nvidia/cuda nvidia-smi
nvidia-docker | 2017/10/10 16:12:44 Error: Could not load UVM kernel module. Is nvidia-modprobe installed?
$ nvidia-smi
Tue Oct 10 16:12:51 2017
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 375.66                 Driver Version: 375.66                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla P100-PCIE...  Off  | 0000:04:00.0     Off |                    0 |
| N/A   32C    P0    32W / 250W |      0MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla P100-PCIE...  Off  | 0000:05:00.0     Off |                    0 |
| N/A   30C    P0    29W / 250W |      0MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   2  Tesla P100-PCIE...  Off  | 0000:84:00.0     Off |                    0 |
| N/A   32C    P0    31W / 250W |      0MiB / 16276MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+
|   3  Tesla P100-PCIE...  Off  | 0000:85:00.0     Off |                    0 |
| N/A   37C    P0    28W / 250W |      0MiB / 16276MiB |      3%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID  Type  Process name                               Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

**Comment by 3XX0** [comment]
Closing as it most likely a driver installation issue and/or fixed in 2.0/master.
